### DISCUSSION TITLE
FDS parallel run fails on windows installtion

### DISCUSSION THREAD

**Discussion by Muthu-V** [firemodels/fds#13790] [General]
Dear All, 
Anyone faced this error and fixed it  ? the same machine in linux mode can run the same case  without fail . 

[mpiexec@PC-Z] wmain (mpiexec.c:2165): assert (pg->intel.exitcodes != NULL) failed
[mpiexec@PC-Z] HYD_sock_write (..\windows\src\hydra_sock.c:387): write error (errno = 34)
[runfds.exe] subprocess exit: -1

Thanks

**Comment by drjfloyd** [comment]
Are you following the instructions in the User's Guide for running FDS under Windows (using CMDfds and fds_local)?

**Reply by Muthu-V** [reply]
Hi Floyd, 
Thank you for your inputs . Here are the commands that was used and the Machine spec and OS. 
fds_local -p 8 -o 4 case.fds
[mpiexec@PC-CZC4157D2Z] wmain (mpiexec.c:2165): assert (pg->intel.exitcodes != NULL) failed
[mpiexec@PC-CZC4157D2Z] HYD_sock_write (..\windows\src\hydra_sock.c:387): write error (errno = 34)

machine Spec: AMD Ryzen Threadripper PRO 7985WX 64-Cores        3.20 GHz
Edition	Windows 11 Enterprise
Version	23H2
Installed on	‎25/‎04/‎2024
OS build	22631.4317

**Reply by mcgratta** [reply]
Do this
```
mpiexec -n 32 hostname
```
This just tests if mpiexec is working, which is the program that handles the multi-processes.

**Reply by Muthu-V** [reply]
C:\Users\505402>mpiexec -n 32 PC-CZC4157D2Z
[mpiexec@PC-CZC4157D2Z] wmain (mpiexec.c:2165): assert (pg->intel.exitcodes != NULL) failed
[mpiexec@PC-CZC4157D2Z] HYD_sock_write (..\windows\src\hydra_sock.c:387): write error (errno = 34)

**Reply by mcgratta** [reply]
I meant for you to type exactly what I typed
```
mpiexec -n 32 hostname
```
`hostname` is a program that will tell you the name of your computer.

**Reply by Muthu-V** [reply]
C:\Users\505402>mpiexec -n 32 hostname
[mpiexec@PC-CZC4157D2Z] wmain (mpiexec.c:2165): assert (pg->intel.exitcodes != NULL) failed
[mpiexec@PC-CZC4157D2Z] HYD_sock_write (..\windows\src\hydra_sock.c:387): write error (errno = 34)

**Reply by mcgratta** [reply]
Something is very wrong here. Are you using the `CMDfds` command prompt on a Windows computer?

**Reply by Muthu-V** [reply]
mpiexec -n 42 fds case.fds works fine . There is an issue with the number of cores that we set for the analysis. 42 cores are working fine . But when we set to 4 cores or 8 cores it didn't  work and gives the above message.

**Reply by Muthu-V** [reply]
Yes, I use CMDfds to execute fds commands on windows

**Reply by Muthu-V** [reply]
By running the command mpiexec -n X hostname where  X= 2, 3, 5, 6, 7, 9, 10, 11, 13, 14, 17, 18, 21, 25, 41 and 42  , works fine both in linux and windows fds solvers. Other mpi core settings throws up the error as mentioned above . Don't know the reason why intel is not supporting this combination. When I tried commercial CFD codes , msmpi parallel settings are working fine , but when I set intelmpi settings it crashes and gave me the same error above. If anyone had seen and fixed it , they may share the workaround. Thanks

**Reply by mcgratta** [reply]
Do you mean that if you type
```
mpiexec -n 4 hostname
```
You get an error? On my laptop, whose name is ash, I see
![image](https://github.com/user-attachments/assets/6d54d406-b68d-4259-b516-9ec99e930228)

**Reply by Muthu-V** [reply]
C:\Users\505402>mpiexec -n 4 hostname
[mpiexec@PC-CZC4157D2Z] wmain (mpiexec.c:2165): assert (pg->intel.exitcodes != NULL) failed
[mpiexec@PC-CZC4157D2Z] HYD_sock_write (..\windows\src\hydra_sock.c:387): write error (errno = 34)

**Reply by Muthu-V** [reply]
if -n = those number series mentioned above, no error it shows up. It also runs in mpiexec .  For example if I set n = 3 , it gave the output like below. 
C:\Users\505402>mpiexec -n 3 hostname
PC-CZC4157D2Z
PC-CZC4157D2Z
PC-CZC4157D2Z

**Reply by mcgratta** [reply]
I have no idea what could be causing this. My only remaining suggestion is from [The IT Crowd](https://www.youtube.com/watch?v=nn2FB1P_Mn8)

**Reply by drjfloyd** [reply]
Since the PC has an AMD Ryzen Threadripper, maybe something with the architecture isn't playing well with Intel MPI or Windows?  If you open the Windows task manager and look at the CPU under the performance tab do you see all the physical cores there?  Do you have any virtualization enabled? If so disabling might help.

**Reply by Muthu-V** [reply]
thank you Kevin and Floyd for your insights . I have to keep the existing mpi core combinations for running the sims locally.  I see the total logical cores 128 numbers , don't see separately physical cores in graphical window, although it mentions that it has 64 cores 
 Haven't tried to disable  the virtualization using BIOS settings . 
![Screenshot 2024-11-24 113544](https://github.com/user-attachments/assets/92de9b31-672b-46eb-b1e3-71e6e453d905)

**Reply by mcgratta** [reply]
We turn off "hyper-threading" in which one uses both logical processors per core. I would turn the "virtualization" off and run jobs with as many as 64 meshes.

**Reply by Muthu-V** [reply]
Thank you Kevin . I Will try this later. Hyperthreading is required for other jobs  that I do. Changing it from BIOS mode every time is not practical .  Perhaps windows should allow the changes by simple clicks on the windows .
